server/app/services/export/QuestionJsonSampler.java

```diff
@@ -0,0 +1,525 @@
+package services.export;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+import static services.export.JsonExporter.exportEntriesToJsonApplication;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import java.util.Optional;
+import javax.inject.Inject;
+import services.CfJsonDocumentContext;
+import services.Path;
+import services.applicant.ApplicantData;
+import services.applicant.question.AddressQuestion;
+import services.applicant.question.ApplicantQuestion;
+import services.applicant.question.CurrencyQuestion;
+import services.applicant.question.DateQuestion;
+import services.applicant.question.EmailQuestion;
+import services.applicant.question.FileUploadQuestion;
+import services.applicant.question.IdQuestion;
+import services.applicant.question.MultiSelectQuestion;
+import services.applicant.question.NameQuestion;
+import services.applicant.question.NumberQuestion;
+import services.applicant.question.PhoneQuestion;
+import services.applicant.question.Question;
+import services.applicant.question.SingleSelectQuestion;
+import services.applicant.question.TextQuestion;
+import services.program.ProgramQuestionDefinition;
+import services.question.LocalizedQuestionOption;
+import services.question.types.QuestionDefinition;
+import services.question.types.QuestionType;
+
+/**
+ * A {@link QuestionJsonSampler} for a Question {@link Q} provides a strategy for showing the
+ * question's answers in JSON form.
+ *
+ * <p>Some {@link QuestionType}s share the same {@link QuestionJsonSampler}.
+ */
+public interface QuestionJsonSampler<Q extends Question> {
+
+  default CfJsonDocumentContext getSampleJson(QuestionDefinition questionDefinition) {
+    if (questionDefinition.getEnumeratorId().isPresent()) {
+      // TODO(#5238): support enumerated questions.
+      return new CfJsonDocumentContext();
+    }
+
+    ProgramQuestionDefinition programQuestionDefinition =
+        ProgramQuestionDefinition.create(questionDefinition, Optional.empty());
+    ApplicantData applicantData = new ApplicantData();
+    ApplicantQuestion applicantQuestion =
+        new ApplicantQuestion(programQuestionDefinition, applicantData, Optional.empty());
+
+    Q question = getQuestion(applicantQuestion);
+    addSampleData(applicantData, question);
+
+    @SuppressWarnings("unchecked")
```

Comment:
Can we add a comment explaining why we're suppressing the warning?

Reply:
Done.